File: Data Structures & Algorithms/duplicate-integer/submission-3.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        std::map<int, int> dict;

        for (const int num : nums){
            if (dict.count(num) == 0){
                dict[num] = 1;
            } else return true;
        }
        return false;
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-3.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap< Integer, Integer > complements = new HashMap<>();

        for (int idx = 0; idx < nums.length; idx++){
            Integer complement = target - nums[idx];
            if (complements.containsKey(complement)){
                return new int[]{complements.get(complement), idx};
            }
            complements.put(nums[idx], idx);
        }

        return new int[]{-1, -1};
        
    }
}
